src/rdkafka_topic.c

```diff
@@ -619,7 +619,7 @@ static int rd_kafka_toppar_leader_update (rd_kafka_topic_t *rkt,
         if (leader != NULL &&
             rktp->rktp_broker != NULL &&
             rktp->rktp_broker->rkb_source != RD_KAFKA_INTERNAL &&
-            rktp->rktp_broker != leader &&
+            rktp->rktp_broker != rktp->rktp_leader &&
```

Comment:
i reasoned it was the same, and thought it was clearer, but i have reverted it now in case i'm missing something (i thought about it during the fff work a lot more than now).

rather than add a comment, i've added an intermediate boolean `fetching_from_follower`, which i think makes it easy enough to follow the if statement (and better than a comment).